framework/encode/dx12_state_writer.cpp

```diff
@@ -1373,26 +1373,44 @@ void Dx12StateWriter::WriteEnableDRED()
 void Dx12StateWriter::WriteAccelerationStructuresState(const Dx12StateTable& state_table)
 {
     std::map<uint64_t, const DxAccelerationStructureBuildInfo*> build_infos;
+    ID3D12Device_Wrapper*                                       device_wrapper = nullptr;
 
     // Find all acceleration structures that exist on resources.
+    uint64_t max_inputs_buffer_size = 0;
     state_table.VisitWrappers([&](ID3D12Resource_Wrapper* resource_wrapper) {
         GFXRECON_ASSERT(resource_wrapper != nullptr);
         GFXRECON_ASSERT(resource_wrapper->GetObjectInfo() != nullptr);
 
         const auto resource_info = resource_wrapper->GetObjectInfo();
 
+        if (device_wrapper == nullptr)
+        {
+            device_wrapper = resource_info->device_wrapper;
```

Comment:
This happened?

Should we log a `GFXRECON_LOG_WARNING`?

Reply:
The logic here is to set `device_wrapper` once in the loop so that we have a `ID3D12Device_Wrapper` for use in constructing a `Dx12ResourceDataUtil` after the loop. I don't think `resource_info->device_wrapper` should ever be null. Line 1401 below ensures a device wrapper was found with an assert.